service/src/main/java/bio/terra/workspace/common/utils/AzureUtils.java

```diff
@@ -0,0 +1,101 @@
+package bio.terra.workspace.common.utils;
+
+import bio.terra.common.exception.ValidationException;
+import bio.terra.workspace.generated.model.ApiAzureVmCustomScriptExtensionSetting;
+import bio.terra.workspace.generated.model.ApiAzureVmCustomScriptExtensionTag;
+import bio.terra.workspace.generated.model.ApiAzureVmImage;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+
+/** Various utilities for validating requests in Azure VM Controllers. */
+public final class AzureUtils {
```

Comment:
AzureVMUtils?

Reply:
fixed the comment. It has non-VM utils as well